### DISCUSSION TITLE
Ipsec tunnels

### DISCUSSION THREAD

**Discussion by dev-daniil** [strongswan/strongswan#1750] [Q&A]
Recently ran into a problem. I had a VPN deployed with two pools, one standard and one with an ad blocker (other DNS). The question arose with the recognition of which of the pools the client needs to connect to. Decided to use the rightid parameter. For a regular connection, I left %any, and for a connection with an ad blocker, I set *@without-ads and noticed such an interesting thing. When connected with Android, everything works fine. When connecting from iOS, always connects only to the pool with rightid = %any. Why does it work like this? Yes, I understand that the implementation of the distribution is not the best, but in theory, it should have worked, and the most interesting thing works, but only on Android. Are there any other ways to achieve the same result as on Android, but for all platforms? (Unfortunately, I cannot remove %any, since this server works in parallel in another application and serves other clients, and I am not allowed to make changes there)

**Comment by tobiasbrunner** [comment]
Read the logs and check which identities are sent and matched etc. My guess is that iOS doesn't send the username as IKE identity so it can't be matched at that point. If so, you need connection switching based on EAP identities, which currently isn't properly supported (see e.g. #1326 where a workaround is discussed).
